core/FObject.js

```diff
@@ -463,8 +463,10 @@ var FObject = {
         continue;
       }
 
-      if ( property.f(this).compareTo(property.f(other)) !== 0) {
-        diff[property.name] = property.f(other);
+      var thisValue = property.f(this);
+      var otherValue = property.f(other);
+      if ( thisValue.compareTo && thisValue.compareTo(otherValue) !== 0) {
+        diff[property.name] = otherValue;
```

Comment:
Slightly better:

 var thisValue = property.f(this);
if ( thisValue.compareTo ) {
    var otherValue = property.f(other);
    if ( thisValue.compareTo(otherValue) !== 0 ) diff[property.name] = otherValue;
}